node/node.go

```diff
@@ -237,32 +244,28 @@ func init() {
 	grpclog = grpc_logsettable.ReplaceGrpcLoggerV2()
 }
 
-func loadConfig(c *cobra.Command) (*config.Config, error) {
-	conf, err := LoadConfigFromFile()
-	if err != nil {
-		return nil, err
-	}
-	if err := cmd.EnsureCLIFlags(c, conf); err != nil {
-		return nil, fmt.Errorf("mapping cli flags to config: %w", err)
+// loadConfig loads config and preset (if provided) into the provided config.
+// It first loads the preset and then overrides it with values from the config file.
+func loadConfig(cfg *config.Config, preset, path string) error {
```

Comment:
Instead of passing in `cfg` with the mainnet settings, we could just use "mainnet" as `preset` if it is not set and return the `*config.Config`.

Reply:
This wouldn't work - I need to pass the config instance that was used to record the flags (in `AddFlags(&cfg)`).

The logic is as follows:
1. create a config instance `cfg` with mainnet defaults
2. add flags, assigning their _future_ values to fields in `cfg` (the future values will be written in p.5)
3. check for `--preset` in CLI args or "preset" field in config file; load preset if any is set (prioritizing `--preset`)
4. load config file
5. apply CLI args (calling `c.ParseFlags()`)